WebService/Auth/AuthMiddleware.cs

```
@@ -202,20 +194,42 @@ private bool ValidateToken(string token, HttpContext context)
             return false;
         }
 
-        private ICollection<SecurityKey> GetSigningKeys()
+        private async Task<bool> InitializeTokenValidationAsync(CancellationToken token)
         {
+            if (this.tokenValidationInitialized) return true;
+
             try
             {
-                return this.openIdCfgMan
-                    .GetConfigurationAsync(CancellationToken.None).Result
-                    .SigningKeys;
+                var openIdConfig = await this.openIdCfgMan.GetConfigurationAsync(token);
```

How is this working but not the above line w.r.t retries?

---

In the previous implementation, OpenId would be initialized only in the ctor, which is executed when the service starts, and exceptions were swallowed to avoid killing the service.
After this change, the ctor still tries to initialize OpenId and still log errors, but in case of failure, the class will retry on the next web request (i.e. the retry is not managed by a timer if you were wondering)